MediaBrowser.Model/Dlna/StreamBuilder.cs

```diff
@@ -1400,7 +1382,9 @@ private static int GetMaxAudioBitrateForTotalBitrate(long totalBitrate)
                         playMethod = PlayMethod.DirectStream;
                     }
 
-                    var ranked = rank(ref failureReasons);
+                    TranscodeReason[] rankings = [TranscodeReason.VideoCodecNotSupported, VideoCodecReasons, TranscodeReason.AudioCodecNotSupported, AudioCodecReasons, ContainerReasons];
```

Comment:
Will this array be re-allocated each iteration? Or will it be considered const?
I would move it out of `Select`.

Generally speaking, I would leave the ranking untouched (except for `new[]`) and refactor it in the next PR. Current PR keeps growing.